dashboard/app/controllers/pd/workshop_material_orders_controller.rb

```diff
@@ -36,9 +40,29 @@ def admin_index
       order_errors: @workshop_material_orders.with_order_errors.count
     }
 
+    # Search emails
+    @workshop_material_orders = @workshop_material_orders.search_emails(params[:email]) if params[:email]
+
+    # filter: [ordered | shipped | errors]
+    if params[:filter]
+      case params[:filter].try(:downcase)
+        when 'ordered'
+          @workshop_material_orders = @workshop_material_orders.successfully_ordered
+        when 'shipped'
+          @workshop_material_orders = @workshop_material_orders.shipped
+        when 'errors'
+          @workshop_material_orders = @workshop_material_orders.with_order_errors
+        else
+          params.delete :filter
+      end
+    end
+
+    @workshop_material_orders = @workshop_material_orders.page(page).per(page_size)
     view_options(full_width: true)
   end
 
+  private
+
   def workshop_material_order_params
```

Comment:
Not sure if this is standard practice, but should we also have a param helper method here to filter out params other than email / filter and such?